ehr/src/org/labkey/ehr/EHRController.java

```diff
@@ -141,9 +143,46 @@ public ApiResponse execute(GetDataEntryItemsForm form, BindException errors)
 
             Collection<DataEntryForm> forms = DataEntryManager.get().getForms(getContainer(), getUser());
             List<JSONObject> formJson = new ArrayList<>();
-            for (DataEntryForm def : forms)
+
+            // restrict forms if the user is in the restricted group
+            var userGroups = SecurityManager.getGroups(getContainer(), getUser());
+            var permittedForms = EHRService.get().getDataEntryFormAccess(getContainer());
+            var isUserInRestrictedGroup = false;
+            String restrictedGroupName = null;
+
+            for (var entry : permittedForms.entrySet())
+            {
+                for (Group g : userGroups)
+                {
+                    if (g.getName().equalsIgnoreCase(entry.getKey()))
+                    {
+                        // assuming user is present in at most 1 restricted group
+                        isUserInRestrictedGroup = true;
+                        restrictedGroupName = entry.getKey();
+                        break;
+                    }
+                }
+            }
+
+            if (isUserInRestrictedGroup)
             {
-                formJson.add(def.toJSON(form.isIncludeFormElements()));
+                for (DataEntryForm f : forms)
+                {
+                    permittedForms.get(restrictedGroupName).forEach(permForm -> {
```

Comment:
I suggest matching on the getName() for the form instead of its class. It's possible that a single class could be registered with different configurations that produce different names